cudax/include/cuda/experimental/__hierarchy/level_dimensions.cuh

```diff
@@ -64,15 +70,25 @@ struct dimensions_handler<dim3>
 };
 
 template <typename Dims, Dims Val>
-struct dimensions_handler<::cuda::std::integral_constant<Dims, Val>>
+struct dimensions_handler<::cuda::std::integral_constant<Dims, Val>> : public base_dimensions_handler
 {
-  static constexpr bool is_type_supported = true;
-
   _CCCL_NODISCARD _CCCL_HOST_DEVICE static constexpr auto translate(const Dims& d) noexcept
   {
     return dimensions<dimensions_index_type, size_t(d), 1, 1>();
   }
 };
+
+// needs_finalizeation or similar might be a better name
+template <typename Dims>
+inline constexpr bool usable_for_queries = false;
+
+template <typename T, size_t... Extents>
+inline constexpr bool usable_for_queries<dimensions<T, Extents...>> = true;
```

Comment:
Remark: I love variable templates as traits instead of structs. They are shorter and more to the point. They require C++14 though, which is why we don't see many of those around here.